tools/cms-scitoken-init.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+// cms-scitoken-init - Go client for CMS scitoken
+// Author: Valentin Kuznetsov <vkuznet@gmail.com>
+// Build instruction:
+//    go get github.com/vkuznet/x509proxy # get x509proxy package
+//    go build cms-scitoken-init.go       # builds cms-scitoken-init executable
+// Run instructions:
+//    cms-scitoken-init -url <issuer URL> # use -verbose to debug HTTP request/responses
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"os/user"
+
+	"github.com/vkuznet/x509proxy"
+)
+
+type OpenID struct {
+	Issuer        string `json:"issuer"`
+	JwksUri       string `json:"jwks_uri"`
+	TokenEndpoint string `json:"token_endpoint"`
+}
+
+func main() {
+	var rurl string
+	flag.StringVar(&rurl, "url", "", "Issuer url")
+	var verbose bool
+	flag.BoolVar(&verbose, "verbose", false, "verbose output")
+	flag.Parse()
+	token := FetchToken(rurl, verbose)
+	fmt.Println(token)
+}
+
+// client X509 certificates
+func tlsCerts(verbose bool) ([]tls.Certificate, error) {
+	uproxy := os.Getenv("X509_USER_PROXY")
+	uckey := os.Getenv("X509_USER_KEY")
+	ucert := os.Getenv("X509_USER_CERT")
+
+	// check if /tmp/x509up_u$UID exists, if so setup X509_USER_PROXY env
+	u, err := user.Current()
+	if err == nil {
+		fname := fmt.Sprintf("/tmp/x509up_u%s", u.Uid)
+		if _, err := os.Stat(fname); err == nil {
+			uproxy = fname
+		}
+	}
+
+	if uproxy == "" && uckey == "" { // user doesn't have neither proxy or user certs
+		return nil, nil
+	}
+	if verbose {
+		fmt.Println("X509_USER_PROXY", uproxy)
+		fmt.Println("X509_USER_KEY", uckey)
+		fmt.Println("X509_USER_CERT", ucert)
+	}
+	if uproxy != "" {
+		// use local implementation of LoadX409KeyPair instead of tls one
+		x509cert, err := x509proxy.LoadX509Proxy(uproxy)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse X509 proxy: %v", err)
+		}
+		if verbose {
+			fmt.Println("Load x509cert via proxy cert")
+		}
+		certs := []tls.Certificate{x509cert}
+		return certs, nil
+	}
+	x509cert, err := tls.LoadX509KeyPair(ucert, uckey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse user X509 certificate: %v", err)
+	}
+	if verbose {
+		fmt.Println("Load x509cert via user key/cert")
+	}
+	certs := []tls.Certificate{x509cert}
+	return certs, nil
+}
+
+// HttpClient is HTTP client for urlfetch server
+func HttpClient(verbose bool) *http.Client {
+	// get X509 certs
+	certs, err := tlsCerts(verbose)
+	if err != nil {
+		panic(err.Error())
+	}
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{Certificates: certs,
+			InsecureSkipVerify: true},
+	}
+	return &http.Client{Transport: tr}
+}
+
+// run go-routine to periodically obtain rucio token
```

Comment:
```suggestion
```

Reply:
removed this comment since it does not belong here.